### ISSUE TITLE
 cannot find -lgcc

### ISSUE THREAD

**Issue by josephspurrier** [matwachich/iup#2]
I changed to Go 1.6 32 instead of 64. I installed mingw32 using win-builds-1.5.0. I make sure c:\mingw32 was part of my path. It doesn't matter if the lib folder is there or not, same error. The libs I'm using are: iup-3.18_Win32_mingw4_lib

When I build, this is my error log:

```
C:/Users/JosephSpurrier/Desktop/Dev/GoPortWin1.6-r.2-iup/go/bin/go.exe build -i [C:/Users/JosephSpurrier/Desktop/Dev/GoPortWin1.6-r.2-iup/workspace/src/hello]
# github.com/matwachich/iup
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/libgcc.a when searching for -lgcc
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3\libgcc.a when searching for -lgcc
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/libgcc.a when searching for -lgcc
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lgcc
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/libgcc_eh.a when searching for -lgcc_eh
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3\libgcc_eh.a when searching for -lgcc_eh
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/libgcc_eh.a when searching for -lgcc_eh
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lgcc_eh
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/lib/libpthread.dll.a when searching for -lpthread
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/lib/libpthread.a when searching for -lpthread
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/lib\libpthread.a when searching for -lpthread
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/lib/libpthread.dll.a when searching for -lpthread
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/lib/libpthread.a when searching for -lpthread
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lpthread
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/libgcc.a when searching for -lgcc
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3\libgcc.a when searching for -lgcc
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/libgcc.a when searching for -lgcc
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lgcc
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/libgcc_eh.a when searching for -lgcc_eh
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3\libgcc_eh.a when searching for -lgcc_eh
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: skipping incompatible c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/libgcc_eh.a when searching for -lgcc_eh
c:/mingw32/bin/../lib64/gcc/x86_64-w64-mingw32/4.8.3/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lgcc_eh
collect2.exe: error: ld returned 1 exit status
Error: process exited with code 2.
```

**Comment by matwachich** [comment]
Are you using x64 C compiler with x86 static IUP libs? Perhaps this is the problem.
I use Go 1.6 x86 on Win7 64 with TDM-GCC 32 with GCC 5.1.0-dw2 (tdm-1).

I was forced to switch to the dw2 version because of the precompiled IUP libs using it (thus the todo item: learn how to compile IUP from source).

Are you using the latest commit?

See this [link](http://stackoverflow.com/a/14991009)

**Comment by josephspurrier** [comment]
Ah, you're right, I installed the wrong version of the C compiler. Let me try again with the i686 version of mingw instead of x86_64. I'll let you know.

**Comment by josephspurrier** [comment]
Awesome, it's working now! Thanks for your help on this. I'll start testing.
